python/ray/data/_internal/planner/plan_read_op.py

```diff
@@ -55,23 +56,34 @@ def _splitrange(n, k):
     return output
 
 
-def _do_additional_splits(
-    blocks: Iterable[Block], _: TaskContext, additional_output_splits: int
-) -> Iterable[Block]:
-    """Do additional splits to the output blocks of a ReadTask.
-
-    Args:
-      blocks: The input blocks.
-      additional_output_splits: The number of additional splits, must be greater than 1.
-    """
-    assert additional_output_splits > 1
-    for block in blocks:
-        block = BlockAccessor.for_block(block)
-        offset = 0
-        split_sizes = _splitrange(block.num_rows(), additional_output_splits)
-        for size in split_sizes:
-            yield block.slice(offset, offset + size, copy=True)
-            offset += size
+class BuildOutputBlocksWithAdditionalSplit(MapTransformFn):
```

Comment:
This one looks bit awkward right now, let's make splitblock an operator later.

Reply:
yeah, also it's also weird that previously BuildOutputBlock was not the last transform fn and caused the bug.